test/e2e/apps/deployment.go

```diff
@@ -134,6 +142,300 @@ var _ = SIGDescribe("Deployment", func() {
 	})
 	// TODO: add tests that cover deployment.Spec.MinReadySeconds once we solved clock-skew issues
 	// See https://github.com/kubernetes/kubernetes/issues/29229
+
+	ginkgo.It("should run the lifecycle of a Deployment", func() {
```

Comment:
Can we separate this into multiple tests?
Current test is one of big tests by comparing the other tests in this test package and that would make it difficult to investigate issues if they happen.

Reply:
every individual test has setup/tear down cost, and our e2e jobs already trend long. A single coherent test that walks an object through its paces seems better to me.